src/components/main-app/ol/clip-raster.js

```diff
@@ -1,17 +1,23 @@
 import { Vector as VectorLayer } from 'ol/layer';
 import { Vector as VectorSource } from 'ol/source';
-import { TopoJSON } from 'ol/format';
+import { GeoJSON, TopoJSON } from 'ol/format';
 import { getVectorContext } from 'ol/render';
 import { Fill, Style, Stroke } from 'ol/style';
 
 /**
  * @param {*} rasterLayer = Raster layer map instance generated from OL map init
  */
+
+function get_url_extension(url) {
+  return url.split(/[#?]/)[0].split('.').pop().trim();
+}
```

Comment:
What happens if data comes from non-extension API? 
I think some other approach is needed.

Reply:
You must get that data and check whether this is a GeoJSON or TopoJSON. 
And based on that you should pass the map feature.

Reply:
done